security/security.c

```diff
@@ -86,6 +90,45 @@ static __initconst const char * const builtin_lsm_order = CONFIG_LSM;
 static __initdata struct lsm_info **ordered_lsms;
 static __initdata struct lsm_info *exclusive;
 
+/* static slots init */
+#define LSM_FUNC_DEFAULT(NAME)		NAME##_func_default
+#define HOOK_STATIC_CALL(HOOK, NUM)	static_call_##HOOK##_##NUM
+#define HOOK_STATIC_CHECK(HOOK, NUM)	static_check_##HOOK##_##NUM
+
+#define CREATE_STATIC(NUM, NAME)				\
+	DEFINE_STATIC_CALL(HOOK_STATIC_CALL(NAME, NUM),		\
+			   LSM_FUNC_DEFAULT(NAME));		\
+	DEFINE_STATIC_KEY_FALSE(HOOK_STATIC_CHECK(NAME, NUM));
+
+// We need a default function that will not be called so that
+// static_call can infer the expected type
```

Comment:
I just realised if these are never called we can almost certainly put them into a linker section where they don't end up in the actual binary. You could just add a TODO for this for now. I guess from the `noinline` that just using an inline function won't work here and we need a real symbol?